feature: Add link to user profile

In the markdown output, make the username a link to the github users profile

Currently, I have the `pytest 7.4.3` version installed. When I run the `make test` command, I get:
```
ERROR: usage: pytest [options] [file_or_dir] [file_or_dir] [...]
pytest: error: unrecognized arguments: --cov=. --cov-config=.coveragerc --cov-fail-under=80 --cov-report
  inifile: None
  rootdir: /workspace/contributors

make: *** [Makefile:3: test] Error 4
```
Why has the above error occurred? How should I proceed ahead?

Installed `pytest-cov` and ran `make test` command. 
The total test coverage has passed the threshold of 80%. 
The test summary info has 3 failed, 12 passed, 1 warning tests.
```bash
pytest -v --cov=. --cov-config=.coveragerc --cov-fail-under=80 --cov-report term-missing
================================================== test session starts ==================================================
platform linux -- Python 3.12.0, pytest-7.4.3, pluggy-1.3.0 -- /home/gitpod/.pyenv/versions/3.12.0/bin/python3
cachedir: .pytest_cache
rootdir: /workspace/contributors
plugins: cov-4.1.0, anyio-4.0.0, pylama-8.4.1
collected 15 items                                                                                                      

test_auth.py::TestAuth::test_auth_to_github_with_ghe PASSED                                                       [  6%]
test_auth.py::TestAuth::test_auth_to_github_with_token PASSED                                                     [ 13%]
test_auth.py::TestAuth::test_auth_to_github_without_token PASSED                                                  [ 20%]
test_contributor_stats.py::TestContributorStats::test_init PASSED                                                 [ 26%]
test_contributor_stats.py::TestContributorStats::test_is_new_contributor_false PASSED                             [ 33%]
test_contributor_stats.py::TestContributorStats::test_is_new_contributor_true PASSED                              [ 40%]
test_contributor_stats.py::TestContributorStats::test_merge_contributors PASSED                                   [ 46%]
test_contributors.py::TestContributors::test_get_all_contributors_with_organization PASSED                        [ 53%]
test_contributors.py::TestContributors::test_get_all_contributors_with_repository PASSED                          [ 60%]
test_contributors.py::TestContributors::test_get_contributors PASSED                                              [ 66%]
test_contributors.py::TestContributors::test_get_contributors_skip_users_with_no_commits PASSED                   [ 73%]
test_env.py::TestEnv::test_get_env_vars FAILED                                                                    [ 80%]
test_env.py::TestEnv::test_get_env_vars_missing_values PASSED                                                     [ 86%]
test_markdown.py::TestMarkdown::test_write_to_markdown FAILED                                                     [ 93%]
test_markdown.py::TestMarkdown::test_write_to_markdown_with_sponsors FAILED                                       [100%]

======================================================= FAILURES ========================================================
_______________________________________________ TestEnv.test_get_env_vars _______________________________________________

self = <test_env.TestEnv testMethod=test_get_env_vars>, mock_getenv = <MagicMock name='getenv' id='140431789051296'>

    @patch("os.getenv")
    def test_get_env_vars(self, mock_getenv):
        """
        Test the get_env_vars function when all environment variables are set correctly.
        """
        mock_getenv.side_effect = [
            "org",
            "repo,repo2",
            "token",
            "",
            "2022-01-01",
            "2022-12-31",
            "False",
        ]
    
        (
            organization,
            repository_list,
            token,
            ghe,
            start_date,
            end_date,
            sponsor_info,
>       ) = env.get_env_vars()

test_env.py:37: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
env.py:67: in get_env_vars
    mention_users = os.getenv("MENTION_USERS", default="true").lower().strip() == "true"
/home/gitpod/.pyenv/versions/3.12.0/lib/python3.12/unittest/mock.py:1134: in __call__
    return self._mock_call(*args, **kwargs)
/home/gitpod/.pyenv/versions/3.12.0/lib/python3.12/unittest/mock.py:1138: in _mock_call
    return self._execute_mock_call(*args, **kwargs)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <MagicMock name='getenv' id='140431789051296'>, args = ('MENTION_USERS',), kwargs = {'default': 'true'}
effect = <list_iterator object at 0x7fb8d2eb6f80>

    def _execute_mock_call(self, /, *args, **kwargs):
        # separate from _increment_mock_call so that awaited functions are
        # executed separately from their call, also AsyncMock overrides this method
    
        effect = self.side_effect
        if effect is not None:
            if _is_exception(effect):
                raise effect
            elif not _callable(effect):
>               result = next(effect)
E               StopIteration

/home/gitpod/.pyenv/versions/3.12.0/lib/python3.12/unittest/mock.py:1195: StopIteration
__________________________________________ TestMarkdown.test_write_to_markdown __________________________________________

self = <test_markdown.TestMarkdown testMethod=test_write_to_markdown>
mock_file = <MagicMock name='open' id='140431795130816'>

    @patch("builtins.open", new_callable=mock_open)
    def test_write_to_markdown(self, mock_file):
        """
        Test the write_to_markdown function.
        """
        person1 = contributor_stats.ContributorStats(
            "user1",
            False,
            "url",
            100,
            "commit url",
            "sponsor_url_1",
        )
        person2 = contributor_stats.ContributorStats(
            "user2",
            False,
            "url2",
            200,
            "commit url2",
            "sponsor_url_2",
        )
        # Set person2 as a new contributor since this cannot be set on initiatization of the object
        person2.new_contributor = True
        collaborators = [
            person1,
            person2,
        ]
    
        write_to_markdown(
            collaborators,
            "filename",
            "2023-01-01",
            "2023-01-02",
            None,
            "org/repo",
            "false",
        )
    
        mock_file.assert_called_once_with("filename", "w", encoding="utf-8")
        mock_file().write.assert_any_call("# Contributors\n\n")
        mock_file().write.assert_any_call(
            "- Date range for contributor list:  2023-01-01 to 2023-01-02\n"
        )
        mock_file().write.assert_any_call(
            "| Total Contributors | Total Contributions | % New Contributors |\n| --- | --- | --- |\n| 2 | 300 | 50.0% |\n\n"
        )
>       mock_file().write.assert_any_call(
            "| Username | Contribution Count | New Contributor | Commits |\n"
            "| --- | --- | --- | --- |\n"
            "| user1 | 100 | False | commit url |\n"
            "| user2 | 200 | True | commit url2 |\n"
        )

test_markdown.py:60: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <MagicMock name='open().write' id='140431807497376'>
args = ('| Username | Contribution Count | New Contributor | Commits |\n| --- | --- | --- | --- |\n| user1 | 100 | False | commit url |\n| user2 | 200 | True | commit url2 |\n',)
kwargs = {}
expected = call('| Username | Contribution Count | New Contributor | Commits |\n| --- | --- | --- | --- |\n| user1 | 100 | False | commit url |\n| user2 | 200 | True | commit url2 |\n')
cause = None
actual = [call('# Contributors\n\n'), call('- Date range for contributor list:  2023-01-01 to 2023-01-02\n'), call('- Repositor...ts |\n| --- | --- | --- | --- |\n| @user1 | 100 | False | commit url |\n| @user2 | 200 | True | commit url2 |\n'), ...]
expected_string = "write('| Username | Contribution Count | New Contributor | Commits |\\n| --- | --- | --- | --- |\\n| user1 | 100 | False | commit url |\\n| user2 | 200 | True | commit url2 |\\n')"

    def assert_any_call(self, /, *args, **kwargs):
        """assert the mock has been called with the specified arguments.
    
        The assert passes if the mock has *ever* been called, unlike
        `assert_called_with` and `assert_called_once_with` that only pass if
        the call is the most recent one."""
        expected = self._call_matcher(_Call((args, kwargs), two=True))
        cause = expected if isinstance(expected, Exception) else None
        actual = [self._call_matcher(c) for c in self.call_args_list]
        if cause or expected not in _AnyComparer(actual):
            expected_string = self._format_mock_call_signature(args, kwargs)
>           raise AssertionError(
                '%s call not found' % expected_string
            ) from cause
E           AssertionError: write('| Username | Contribution Count | New Contributor | Commits |\n| --- | --- | --- | --- |\n| user1 | 100 | False | commit url |\n| user2 | 200 | True | commit url2 |\n') call not found

/home/gitpod/.pyenv/versions/3.12.0/lib/python3.12/unittest/mock.py:1015: AssertionError
___________________________________ TestMarkdown.test_write_to_markdown_with_sponsors ___________________________________

self = <test_markdown.TestMarkdown testMethod=test_write_to_markdown_with_sponsors>
mock_file = <MagicMock name='open' id='140431789408176'>

    @patch("builtins.open", new_callable=mock_open)
    def test_write_to_markdown_with_sponsors(self, mock_file):
        """
        Test the write_to_markdown function with sponsors info turned on.
        """
        person1 = contributor_stats.ContributorStats(
            "user1",
            False,
            "url",
            100,
            "commit url",
            "sponsor_url_1",
        )
        person2 = contributor_stats.ContributorStats(
            "user2",
            False,
            "url2",
            200,
            "commit url2",
            "",
        )
        # Set person2 as a new contributor since this cannot be set on initiatization of the object
        person2.new_contributor = True
        collaborators = [
            person1,
            person2,
        ]
    
        write_to_markdown(
            collaborators,
            "filename",
            "2023-01-01",
            "2023-01-02",
            None,
            "org/repo",
            "true",
        )
    
        mock_file.assert_called_once_with("filename", "w", encoding="utf-8")
        mock_file().write.assert_any_call("# Contributors\n\n")
        mock_file().write.assert_any_call(
            "- Date range for contributor list:  2023-01-01 to 2023-01-02\n"
        )
        mock_file().write.assert_any_call(
            "| Total Contributors | Total Contributions | % New Contributors |\n| --- | --- | --- |\n| 2 | 300 | 50.0% |\n\n"
        )
>       mock_file().write.assert_any_call(
            "| Username | Contribution Count | New Contributor | Sponsor URL | Commits |\n"
            "| --- | --- | --- | --- | --- |\n"
            "| user1 | 100 | False | [Sponsor Link](sponsor_url_1) | commit url |\n"
            "| user2 | 200 | True | not sponsorable | commit url2 |\n"
        )

test_markdown.py:113: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <MagicMock name='open().write' id='140431790547520'>
args = ('| Username | Contribution Count | New Contributor | Sponsor URL | Commits |\n| --- | --- | --- | --- | --- |\n| user...100 | False | [Sponsor Link](sponsor_url_1) | commit url |\n| user2 | 200 | True | not sponsorable | commit url2 |\n',)
kwargs = {}
expected = call('| Username | Contribution Count | New Contributor | Sponsor URL | Commits |\n| --- | --- | --- | --- | --- |\n| ... 100 | False | [Sponsor Link](sponsor_url_1) | commit url |\n| user2 | 200 | True | not sponsorable | commit url2 |\n')
cause = None
actual = [call('# Contributors\n\n'), call('- Date range for contributor list:  2023-01-01 to 2023-01-02\n'), call('- Repositor...False | [Sponsor Link](sponsor_url_1) | commit url |\n| @user2 | 200 | True | not sponsorable | commit url2 |\n'), ...]
expected_string = "write('| Username | Contribution Count | New Contributor | Sponsor URL | Commits |\\n| --- | --- | --- | --- | --- |\...0 | False | [Sponsor Link](sponsor_url_1) | commit url |\\n| user2 | 200 | True | not sponsorable | commit url2 |\\n')"

    def assert_any_call(self, /, *args, **kwargs):
        """assert the mock has been called with the specified arguments.
    
        The assert passes if the mock has *ever* been called, unlike
        `assert_called_with` and `assert_called_once_with` that only pass if
        the call is the most recent one."""
        expected = self._call_matcher(_Call((args, kwargs), two=True))
        cause = expected if isinstance(expected, Exception) else None
        actual = [self._call_matcher(c) for c in self.call_args_list]
        if cause or expected not in _AnyComparer(actual):
            expected_string = self._format_mock_call_signature(args, kwargs)
>           raise AssertionError(
                '%s call not found' % expected_string
            ) from cause
E           AssertionError: write('| Username | Contribution Count | New Contributor | Sponsor URL | Commits |\n| --- | --- | --- | --- | --- |\n| user1 | 100 | False | [Sponsor Link](sponsor_url_1) | commit url |\n| user2 | 200 | True | not sponsorable | commit url2 |\n') call not found

/home/gitpod/.pyenv/versions/3.12.0/lib/python3.12/unittest/mock.py:1015: AssertionError
=================================================== warnings summary ====================================================
../../home/gitpod/.pyenv/versions/3.12.0/lib/python3.12/site-packages/dateutil/tz/tz.py:37
  /home/gitpod/.pyenv/versions/3.12.0/lib/python3.12/site-packages/dateutil/tz/tz.py:37: DeprecationWarning: datetime.datetime.utcfromtimestamp() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.fromtimestamp(timestamp, datetime.UTC).
    EPOCH = datetime.datetime.utcfromtimestamp(0)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html

---------- coverage: platform linux, python 3.12.0-final-0 -----------
Name                        Stmts   Miss  Cover   Missing
---------------------------------------------------------
auth.py                        10      1    90%   26
contributor_stats.py           46     12    74%   58, 144-178
contributors.py                59     19    68%   16-59, 136, 147-148, 154, 166-169, 175
env.py                         31      8    74%   44, 51, 56, 63, 70-76
markdown.py                    64     11    83%   74, 110, 121-122, 170-177
test_auth.py                   19      1    95%   44
test_contributor_stats.py      31      1    97%   158
test_contributors.py           51      1    98%   152
test_env.py                    22      8    64%   39-45, 60
test_markdown.py               31      1    97%   122
---------------------------------------------------------
TOTAL                         364     63    83%

Required test coverage of 80% reached. Total coverage: 82.69%
================================================ short test summary info ================================================
FAILED test_env.py::TestEnv::test_get_env_vars - StopIteration
FAILED test_markdown.py::TestMarkdown::test_write_to_markdown - AssertionError: write('| Username | Contribution Count | New Contributor | Commits |\n| --- | --- | --- | --- |\n| u...
FAILED test_markdown.py::TestMarkdown::test_write_to_markdown_with_sponsors - AssertionError: write('| Username | Contribution Count | New Contributor | Sponsor URL | Commits |\n| --- | --- | --...
======================================== 3 failed, 12 passed, 1 warning in 1.14s ========================================
```

--- a/.github/linters/.jscpd.json
+++ b/.github/linters/.jscpd.json
@@ -1,5 +1,5 @@
 {
-  "threshold": 25,
+  "threshold": 50,
   "ignore": [
     "test*"
   ],

--- a/README.md
+++ b/README.md
@@ -52,7 +52,7 @@ Below are the allowed configuration options:
 | `START_DATE`          | False    |   Beginning of time      | The date from which you want to start gathering contributor information. ie. Aug 1st, 2023 would be `2023-08-01` If `start_date` and `end_date` are specified then the action will determine if the contributor is new. A new contributor is one that has contributed in the date range specified but not before the start date. **Performance Note:** Using start and end dates will reduce speed of the action by approximately 63X. ie without dates if the action takes 1.7 seconds, it will take 1 minute and 47 seconds.|
 | `END_DATE`            | False    |   Current Date      | The date at which you want to stop gathering contributor information. Must be later than the `START_DATE`. ie. Aug 2nd, 2023 would be `2023-08-02` If `start_date` and `end_date` are specified then the action will determine if the contributor is new. A new contributor is one that has contributed in the date range specified but not before the start date. |
 | `SPONSOR_INFO`        | False    |   False      | If you want to include sponsor information in the output. This will include the sponsor count and the sponsor URL. This will impact action performance. ie. SPONSOR_INFO = "False" or SPONSOR_INFO = "True" |
-| `LINK_TO_PROFILE`     | False    |   True      | If you want to link usernames to their Github profiles in the output. ie. LINK_TO_PROFILE = "True" or LINK_TO_PROFILE = "False" |
+| `LINK_TO_PROFILE`     | False    |   True      | If you want to link usernames to their GitHub profiles in the output. ie. LINK_TO_PROFILE = "True" or LINK_TO_PROFILE = "False" |
 
 ### Example workflows
 

--- a/README.md
+++ b/README.md
@@ -52,6 +52,7 @@ Below are the allowed configuration options:
 | `START_DATE`          | False    |   Beginning of time      | The date from which you want to start gathering contributor information. ie. Aug 1st, 2023 would be `2023-08-01` If `start_date` and `end_date` are specified then the action will determine if the contributor is new. A new contributor is one that has contributed in the date range specified but not before the start date. **Performance Note:** Using start and end dates will reduce speed of the action by approximately 63X. ie without dates if the action takes 1.7 seconds, it will take 1 minute and 47 seconds.|
 | `END_DATE`            | False    |   Current Date      | The date at which you want to stop gathering contributor information. Must be later than the `START_DATE`. ie. Aug 2nd, 2023 would be `2023-08-02` If `start_date` and `end_date` are specified then the action will determine if the contributor is new. A new contributor is one that has contributed in the date range specified but not before the start date. |
 | `SPONSOR_INFO`        | False    |   False      | If you want to include sponsor information in the output. This will include the sponsor count and the sponsor URL. This will impact action performance. ie. SPONSOR_INFO = "False" or SPONSOR_INFO = "True" |
+| `LINK_TO_PROFILE`     | False    |   True      | If you want to link usernames to their Github profiles in the output. ie. LINK_TO_PROFILE = "True" or LINK_TO_PROFILE = "False" |
 
 ### Example workflows
 
@@ -117,7 +118,7 @@ jobs:
 
 | Username | Contribution Count | New Contributor | Commits |
 | --- | --- | --- | --- |
-| zkoppert | 143 | False | [super-linter/super-linter](https://github.com/super-linter/super-linter/commits?author=zkoppert&since=2021-01-01&until=2023-10-10) |
+| @zkoppert | 143 | False | [super-linter/super-linter](https://github.com/super-linter/super-linter/commits?author=zkoppert&since=2021-01-01&until=2023-10-10) |
 
 ## Example Markdown output with no dates supplied
 
@@ -131,7 +132,7 @@ jobs:
 
 | Username | Contribution Count | New Contributor | Sponsor URL | Commits |
 | --- | --- | --- | --- | --- |
-| zkoppert | 1913 | False | [Sponsor Link](https://github.com/sponsors/zkoppert) | [super-linter/super-linter](https://github.com/super-linter/super-linter/commits?author=zkoppert&since=2021-09-01&until=2023-09-30) |
+| @zkoppert | 1913 | False | [Sponsor Link](https://github.com/sponsors/zkoppert) | [super-linter/super-linter](https://github.com/super-linter/super-linter/commits?author=zkoppert&since=2021-09-01&until=2023-09-30) |
 
 ## Local usage without Docker
 

--- a/contributors.py
+++ b/contributors.py
@@ -21,6 +21,7 @@ def main():
         start_date,
         end_date,
         sponsor_info,
+        ltp
     ) = env.get_env_vars()
 
     # Auth to GitHub.com
@@ -64,6 +65,7 @@ def main():
         organization,
         repository_list,
         sponsor_info,
+        ltp
     )
     # write_to_json(contributors)
 

--- a/contributors.py
+++ b/contributors.py
@@ -21,7 +21,7 @@ def main():
         start_date,
         end_date,
         sponsor_info,
-        ltp
+        link_to_profile,
     ) = env.get_env_vars()
 
     # Auth to GitHub.com
@@ -65,7 +65,7 @@ def main():
         organization,
         repository_list,
         sponsor_info,
-        ltp
+        link_to_profile,
     )
     # write_to_json(contributors)
 

--- a/env.py
+++ b/env.py
@@ -11,7 +11,7 @@ from dotenv import load_dotenv
 
 
 def get_env_vars() -> (
-    tuple[str | None, list[str], str, str, str | None, str | None, str | Any]
+    tuple[str | None, list[str], str, str, str | None, str | None, str | Any, str | Any]
 ):
     """
     Get the environment variables for use in the action.
@@ -27,6 +27,7 @@ def get_env_vars() -> (
         str: the start date to get contributor information from
         str: the end date to get contributor information to.
         str: whether to get sponsor information on the contributor
+        str: whether to link username to Github profile in markdown output
 
     """
     # Load from .env file if it exists
@@ -67,6 +68,15 @@ def get_env_vars() -> (
                 "SPONSOR_INFO environment variable not a boolean. ie. True or False or blank"
             )
 
+    ltp = os.getenv("LINK_TO_PROFILE")
+    # make sure that ltp is a boolean
+    if ltp:
+        ltp = ltp.lower().strip()
+        if ltp not in ["true", "false", ""]:
+            raise ValueError(
+                "LINK_TO_PROFILE environment variable not a boolean. ie. True or False or blank"
+            )
+
     # Separate repositories_str into a list based on the comma separator
     repositories_list = []
     if repositories_str:
@@ -82,4 +92,5 @@ def get_env_vars() -> (
         start_date,
         end_date,
         sponsor_info,
+        ltp,
     )

--- a/env.py
+++ b/env.py
@@ -68,11 +68,11 @@ def get_env_vars() -> (
                 "SPONSOR_INFO environment variable not a boolean. ie. True or False or blank"
             )
 
-    ltp = os.getenv("LINK_TO_PROFILE")
-    # make sure that ltp is a boolean
-    if ltp:
-        ltp = ltp.lower().strip()
-        if ltp not in ["true", "false", ""]:
+    link_to_profile = os.getenv("LINK_TO_PROFILE")
+    # make sure that link_to_profile is a boolean
+    if link_to_profile:
+        link_to_profile = link_to_profile.lower().strip()
+        if link_to_profile not in ["true", "false", ""]:
             raise ValueError(
                 "LINK_TO_PROFILE environment variable not a boolean. ie. True or False or blank"
             )
@@ -92,5 +92,5 @@ def get_env_vars() -> (
         start_date,
         end_date,
         sponsor_info,
-        ltp,
+        link_to_profile,
     )

--- a/markdown.py
+++ b/markdown.py
@@ -10,6 +10,7 @@ def write_to_markdown(
     organization,
     repository,
     sponsor_info,
+    ltp
 ):
     """
     This function writes a list of collaborators to a markdown file in table format.
@@ -24,6 +25,7 @@ def write_to_markdown(
         organization (str): The organization for which the contributors are being listed.
         repository (str): The repository for which the contributors are being listed.
         sponsor_info (str): True if the user wants the sponsor_url shown in the report
+        ltp (str): True if the user wants the username linked to Github profile in the report
 
     Returns:
         None
@@ -31,7 +33,7 @@ def write_to_markdown(
     """
     # Put together the contributor table
     table, total_contributions = get_contributor_table(
-        collaborators, start_date, end_date, organization, repository, sponsor_info
+        collaborators, start_date, end_date, organization, repository, sponsor_info, ltp
     )
 
     # Put together the summary table including # of new contributions, # of new contributors, % new contributors, % returning contributors
@@ -127,7 +129,7 @@ def get_summary_table(collaborators, start_date, end_date, total_contributions):
 
 
 def get_contributor_table(
-    collaborators, start_date, end_date, organization, repository, sponsor_info
+    collaborators, start_date, end_date, organization, repository, sponsor_info, ltp
 ):
     """
     This function returns a string containing a markdown table of the contributors and the total contribution count.
@@ -140,6 +142,7 @@ def get_contributor_table(
         organization (str): The organization for which the contributors are being listed.
         repository (str): The repository for which the contributors are being listed.
         sponsor_info (str): True if the user wants the sponsor_url shown in the report
+        ltp (str): True if the user wants the username linked to Github profile in the report
 
     Returns:
         table (str): A string containing a markdown table of the contributors and the total contribution count.
@@ -177,7 +180,7 @@ def get_contributor_table(
                 commit_urls += url + ", "
         new_contributor = collaborator.new_contributor
 
-        row = f"| {username} | {contribution_count} |"
+        row = f"| {'' if ltp == 'false' else '@'}{username} | {contribution_count} |"
         if "New Contributor" in columns:
             row += f" {new_contributor} |"
         if "Sponsor URL" in columns:

--- a/markdown.py
+++ b/markdown.py
@@ -33,7 +33,13 @@ def write_to_markdown(
     """
     # Put together the contributor table
     table, total_contributions = get_contributor_table(
-        collaborators, start_date, end_date, organization, repository, sponsor_info, link_to_profile
+        collaborators,
+        start_date,
+        end_date,
+        organization,
+        repository,
+        sponsor_info,
+        link_to_profile,
     )
 
     # Put together the summary table including # of new contributions, # of new contributors, % new contributors, % returning contributors
@@ -129,7 +135,13 @@ def get_summary_table(collaborators, start_date, end_date, total_contributions):
 
 
 def get_contributor_table(
-    collaborators, start_date, end_date, organization, repository, sponsor_info, link_to_profile
+    collaborators,
+    start_date,
+    end_date,
+    organization,
+    repository,
+    sponsor_info,
+    link_to_profile,
 ):
     """
     This function returns a string containing a markdown table of the contributors and the total contribution count.

--- a/markdown.py
+++ b/markdown.py
@@ -10,7 +10,7 @@ def write_to_markdown(
     organization,
     repository,
     sponsor_info,
-    ltp
+    link_to_profile,
 ):
     """
     This function writes a list of collaborators to a markdown file in table format.
@@ -25,7 +25,7 @@ def write_to_markdown(
         organization (str): The organization for which the contributors are being listed.
         repository (str): The repository for which the contributors are being listed.
         sponsor_info (str): True if the user wants the sponsor_url shown in the report
-        ltp (str): True if the user wants the username linked to Github profile in the report
+        link_to_profile (str): True if the user wants the username linked to Github profile in the report
 
     Returns:
         None
@@ -33,7 +33,7 @@ def write_to_markdown(
     """
     # Put together the contributor table
     table, total_contributions = get_contributor_table(
-        collaborators, start_date, end_date, organization, repository, sponsor_info, ltp
+        collaborators, start_date, end_date, organization, repository, sponsor_info, link_to_profile
     )
 
     # Put together the summary table including # of new contributions, # of new contributors, % new contributors, % returning contributors
@@ -129,7 +129,7 @@ def get_summary_table(collaborators, start_date, end_date, total_contributions):
 
 
 def get_contributor_table(
-    collaborators, start_date, end_date, organization, repository, sponsor_info, ltp
+    collaborators, start_date, end_date, organization, repository, sponsor_info, link_to_profile
 ):
     """
     This function returns a string containing a markdown table of the contributors and the total contribution count.
@@ -142,7 +142,7 @@ def get_contributor_table(
         organization (str): The organization for which the contributors are being listed.
         repository (str): The repository for which the contributors are being listed.
         sponsor_info (str): True if the user wants the sponsor_url shown in the report
-        ltp (str): True if the user wants the username linked to Github profile in the report
+        link_to_profile (str): True if the user wants the username linked to Github profile in the report
 
     Returns:
         table (str): A string containing a markdown table of the contributors and the total contribution count.
@@ -180,7 +180,7 @@ def get_contributor_table(
                 commit_urls += url + ", "
         new_contributor = collaborator.new_contributor
 
-        row = f"| {'' if ltp == 'false' else '@'}{username} | {contribution_count} |"
+        row = f"| {'' if link_to_profile == 'false' else '@'}{username} | {contribution_count} |"
         if "New Contributor" in columns:
             row += f" {new_contributor} |"
         if "Sponsor URL" in columns:

--- a/test_env.py
+++ b/test_env.py
@@ -24,6 +24,7 @@ class TestEnv(unittest.TestCase):
             "2022-01-01",
             "2022-12-31",
             "False",
+            "True"
         ]
 
         (
@@ -34,6 +35,7 @@ class TestEnv(unittest.TestCase):
             start_date,
             end_date,
             sponsor_info,
+            ltp
         ) = env.get_env_vars()
 
         self.assertEqual(organization, "org")
@@ -43,6 +45,7 @@ class TestEnv(unittest.TestCase):
         self.assertEqual(start_date, "2022-01-01")
         self.assertEqual(end_date, "2022-12-31")
         self.assertEqual(sponsor_info, "false")
+        self.assertEqual(ltp, 'true')
 
     @patch("os.getenv")
     def test_get_env_vars_missing_values(self, mock_getenv):
@@ -50,7 +53,7 @@ class TestEnv(unittest.TestCase):
         Test the get_env_vars function when none of the environment variables are set.
         Expect a ValueError to be raised.
         """
-        mock_getenv.side_effect = [None, None, None, None, None, None, None]
+        mock_getenv.side_effect = [None, None, None, None, None, None, None, None]
 
         with self.assertRaises(ValueError):
             env.get_env_vars()

--- a/test_env.py
+++ b/test_env.py
@@ -35,7 +35,7 @@ class TestEnv(unittest.TestCase):
             start_date,
             end_date,
             sponsor_info,
-            ltp
+            link_to_profile,
         ) = env.get_env_vars()
 
         self.assertEqual(organization, "org")
@@ -45,7 +45,7 @@ class TestEnv(unittest.TestCase):
         self.assertEqual(start_date, "2022-01-01")
         self.assertEqual(end_date, "2022-12-31")
         self.assertEqual(sponsor_info, "false")
-        self.assertEqual(ltp, 'true')
+        self.assertEqual(link_to_profile, 'true')
 
     @patch("os.getenv")
     def test_get_env_vars_missing_values(self, mock_getenv):

--- a/test_env.py
+++ b/test_env.py
@@ -24,7 +24,7 @@ class TestEnv(unittest.TestCase):
             "2022-01-01",
             "2022-12-31",
             "False",
-            "True"
+            "True",
         ]
 
         (
@@ -45,7 +45,7 @@ class TestEnv(unittest.TestCase):
         self.assertEqual(start_date, "2022-01-01")
         self.assertEqual(end_date, "2022-12-31")
         self.assertEqual(sponsor_info, "false")
-        self.assertEqual(link_to_profile, 'true')
+        self.assertEqual(link_to_profile, "true")
 
     @patch("os.getenv")
     def test_get_env_vars_missing_values(self, mock_getenv):

--- a/test_markdown.py
+++ b/test_markdown.py
@@ -47,7 +47,7 @@ class TestMarkdown(unittest.TestCase):
             None,
             "org/repo",
             "false",
-            "true"
+            "true",
         )
 
         mock_file.assert_called_once_with("filename", "w", encoding="utf-8")
@@ -101,7 +101,7 @@ class TestMarkdown(unittest.TestCase):
             None,
             "org/repo",
             "true",
-            "true"
+            "true",
         )
 
         mock_file.assert_called_once_with("filename", "w", encoding="utf-8")
@@ -155,7 +155,7 @@ class TestMarkdown(unittest.TestCase):
             None,
             "org/repo",
             "false",
-            "false"
+            "false",
         )
 
         mock_file.assert_called_once_with("filename", "w", encoding="utf-8")

--- a/test_markdown.py
+++ b/test_markdown.py
@@ -47,6 +47,7 @@ class TestMarkdown(unittest.TestCase):
             None,
             "org/repo",
             "false",
+            "true"
         )
 
         mock_file.assert_called_once_with("filename", "w", encoding="utf-8")
@@ -60,8 +61,8 @@ class TestMarkdown(unittest.TestCase):
         mock_file().write.assert_any_call(
             "| Username | Contribution Count | New Contributor | Commits |\n"
             "| --- | --- | --- | --- |\n"
-            "| user1 | 100 | False | commit url |\n"
-            "| user2 | 200 | True | commit url2 |\n"
+            "| @user1 | 100 | False | commit url |\n"
+            "| @user2 | 200 | True | commit url2 |\n"
         )
 
     @patch("builtins.open", new_callable=mock_open)
@@ -100,6 +101,7 @@ class TestMarkdown(unittest.TestCase):
             None,
             "org/repo",
             "true",
+            "true"
         )
 
         mock_file.assert_called_once_with("filename", "w", encoding="utf-8")
@@ -113,8 +115,62 @@ class TestMarkdown(unittest.TestCase):
         mock_file().write.assert_any_call(
             "| Username | Contribution Count | New Contributor | Sponsor URL | Commits |\n"
             "| --- | --- | --- | --- | --- |\n"
-            "| user1 | 100 | False | [Sponsor Link](sponsor_url_1) | commit url |\n"
-            "| user2 | 200 | True | not sponsorable | commit url2 |\n"
+            "| @user1 | 100 | False | [Sponsor Link](sponsor_url_1) | commit url |\n"
+            "| @user2 | 200 | True | not sponsorable | commit url2 |\n"
+        )
+
+    @patch("builtins.open", new_callable=mock_open)
+    def test_write_to_markdown_without_link_to_profile(self, mock_file):
+        """
+        Test the write_to_markdown function with link to profile turned off.
+        """
+        person1 = contributor_stats.ContributorStats(
+            "user1",
+            False,
+            "url",
+            100,
+            "commit url",
+            "sponsor_url_1",
+        )
+        person2 = contributor_stats.ContributorStats(
+            "user2",
+            False,
+            "url2",
+            200,
+            "commit url2",
+            "sponsor_url_2",
+        )
+        # Set person2 as a new contributor since this cannot be set on initiatization of the object
+        person2.new_contributor = True
+        collaborators = [
+            person1,
+            person2,
+        ]
+
+        write_to_markdown(
+            collaborators,
+            "filename",
+            "2023-01-01",
+            "2023-01-02",
+            None,
+            "org/repo",
+            "false",
+            "false"
+        )
+
+        mock_file.assert_called_once_with("filename", "w", encoding="utf-8")
+        mock_file().write.assert_any_call("# Contributors\n\n")
+        mock_file().write.assert_any_call(
+            "- Date range for contributor list:  2023-01-01 to 2023-01-02\n"
+        )
+        mock_file().write.assert_any_call(
+            "| Total Contributors | Total Contributions | % New Contributors |\n| --- | --- | --- |\n| 2 | 300 | 50.0% |\n\n"
+        )
+        mock_file().write.assert_any_call(
+            "| Username | Contribution Count | New Contributor | Commits |\n"
+            "| --- | --- | --- | --- |\n"
+            "| user1 | 100 | False | commit url |\n"
+            "| user2 | 200 | True | commit url2 |\n"
         )
 
 

